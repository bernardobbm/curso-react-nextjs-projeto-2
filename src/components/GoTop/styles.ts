import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;

    position: fixed;
    bottom: 1.25rem;
    right: 1.25rem;
    z-index: 6;
  `}
`;
