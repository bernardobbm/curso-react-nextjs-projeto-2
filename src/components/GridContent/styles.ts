import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 580px;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.large} 0;
  `}
`;
