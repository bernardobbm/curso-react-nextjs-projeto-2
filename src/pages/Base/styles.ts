import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 3.375rem;

    @media ${theme.media.lteMedium} {
      padding-top: 0;
    }
  `}
`;
