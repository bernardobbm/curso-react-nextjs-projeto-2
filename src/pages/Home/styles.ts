import styled, { css } from 'styled-components';

interface WrapperProps {
  background: string;
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ background }) => css`
    background: ${background};
  `}
`;
