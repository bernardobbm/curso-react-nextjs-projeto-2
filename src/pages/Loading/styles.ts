import styled, { css, keyframes } from 'styled-components';

function rotate() {
  return keyframes`
    0% {
      transform: translate(-50%, -50%) rotate(0deg);
    }

    100% {
      transform: translate(-50%, -50%) rotate(360deg);
    }
  `;
}

export const Container = styled.div`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 10;
    background-color: ${theme.colors.primaryColor};

    &::before,
    &::after {
      content: '';
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      border-width: 0.31rem;
      border-color: transparent;
      border-style: solid;
      border-radius: 50%;
    }

    &::after {
      width: 3.75rem;
      height: 3.75rem;
      border-left: 0.31rem solid ${theme.colors.secondaryColor};
      border-top: 0.31rem solid ${theme.colors.secondaryColor};
      animation: ${rotate()} 0.6s linear infinite;
    }

    &::before {
      width: 1.25rem;
      height: 1.25rem;
      border-left: 0.31rem solid ${theme.colors.secondaryColor};
      border-top: 0.31rem solid ${theme.colors.secondaryColor};
      animation: ${rotate()} 1s linear reverse infinite;
    }
  `}
`;
