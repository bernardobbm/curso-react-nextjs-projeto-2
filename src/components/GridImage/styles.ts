import styled, { css } from 'styled-components';

import { Container as TextComponentContainerStyle } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponentContainerStyle} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 0.3s ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
