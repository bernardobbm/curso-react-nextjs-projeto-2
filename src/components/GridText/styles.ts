import styled, { css } from 'styled-components';

import { Title as HeadingTitleStyle } from '../Heading/styles';
import { Container as TextComponentContainerStyle } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponentContainerStyle} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    ${HeadingTitleStyle} {
      position: relative;
      left: 3.125rem;
    }

    ${HeadingTitleStyle}:before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4.5rem;
      top: -2rem;
      left: -3.8rem;
      transform: rotate(8deg);
    }
  `}
`;
