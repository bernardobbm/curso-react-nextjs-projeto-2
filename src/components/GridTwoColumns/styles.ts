import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;
