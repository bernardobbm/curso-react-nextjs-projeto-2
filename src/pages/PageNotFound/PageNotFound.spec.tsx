import { screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { PageNotFound } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<PageNotFound />', () => {
  it('should render PageNotFound with a Heading and TextComponent', () => {
    renderTheme(<PageNotFound />);

    const heading = screen.getByRole('heading', { name: 'Error 404' });
    const textComponent = screen.getByText(
      'A página que você está buscando não foi encontrada',
    );

    expect(heading).toBeInTheDocument();
    expect(textComponent).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PageNotFound />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
