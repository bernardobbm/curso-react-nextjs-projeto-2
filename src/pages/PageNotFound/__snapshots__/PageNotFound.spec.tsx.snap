// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PageNotFound /> > should match snapshot 1`] = `
.c3 {
  color: #fff;
  font-size: 4rem;
  text-transform: uppercase;
}

.c0 {
  background-color: #fff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0a1128;
  color: #fff;
}

.c1 {
  max-width: 75rem;
  margin: 0 auto;
  padding: 2rem;
  width: 100%;
}

.c5 {
  font-size: 1.5rem;
}

.c2 {
  text-align: center;
  max-width: 580px;
  margin: 0 auto;
}

.c4 {
  margin: 2rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 2.5rem;
  }
}

<div
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Error 404
      </h2>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <p>
            A página que você está buscando não foi encontrada
          </p>
          <br />
          
        
          <a
            href="/"
          >
            Clique aqui para voltar
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
